lib/output.js

```diff
@@ -60,32 +60,35 @@ function getReporter(format) {
   return require('./formats/' + format);
 }
 
-module.exports = function (parameters, response, cb) {
-  var renderer = getReporter(parameters.format);
-  var threshold = parameters.threshold || THRESHOLD;
-  var optimizedResoruceURL = RESOURCE_URL + querystring.stringify({url: response.id, strategy: parameters.strategy})
-
-  console.log(renderer(
-    overview(humanizeUrl(response.id), parameters.strategy, response.ruleGroups.SPEED.score),
-    statistics(response.pageStats),
-    ruleSetResults(response.formattedResults.ruleResults),
-    threshold
-  ));
-
-  if (parameters.optimized) {
-    console.log('\nHere are your optimized images: ', humanizeUrl(optimizedResoruceURL));
-  }
-
-  if (response.ruleGroups.SPEED.score < threshold) {
-    var err = new Error('Threshold of ' + threshold + ' not met with score of ' + response.ruleGroups.SPEED.score);
-    return cb(err);
-  }
-
-  if (parameters.download) {
-      new Download()
-          .get(optimizedResoruceURL)
-          .dest('.')
-          .rename('./optimized.zip')
-          .run(cb);
-  }
+module.exports = function (parameters, response) {
+  return Promise.resolve().then(function () {
+    var renderer = getReporter(parameters.format);
+    var threshold = parameters.threshold || THRESHOLD;
+    var optimizedResoruceURL = RESOURCE_URL + querystring.stringify({url: response.id, strategy: parameters.strategy});
+
+    console.log(renderer(
+      overview(humanizeUrl(response.id), parameters.strategy, response.ruleGroups.SPEED.score),
+      statistics(response.pageStats),
+      ruleSetResults(response.formattedResults.ruleResults),
+      threshold
+    ));
+
+    if (parameters.optimized) {
+      console.log('\nHere are your optimized images: ', humanizeUrl(optimizedResoruceURL));
+    }
+
+    if (response.ruleGroups.SPEED.score < threshold) {
+      var err = new Error('Threshold of ' + threshold + ' not met with score of ' + response.ruleGroups.SPEED.score);
+      throw err;
+    }
+
+    if (parameters.download) {
+      var download = new Download()
+        .get(optimizedResoruceURL)
+        .dest('.')
+        .rename('./optimized.zip');
+
+      return pify(download.run.bind(download), Promise)();
```

Comment:
Fixed it for you https://github.com/addyosmani/psi/commit/ee69ee848f4ddfe52176cadc5e3b7e022d1068ba.